api/proto/teleport/legacy/types/types.proto

```diff
@@ -2428,6 +2442,13 @@ message AccessRequestFilter {
   string User = 2 [(gogoproto.jsontag) = "user,omitempty"];
   // RequestState filters for requests in a specific state.
   RequestState State = 3 [(gogoproto.jsontag) = "state,omitempty"];
+  // SearchKeywords is a list of search keywords to match against resource field values.
+  repeated string SearchKeywords = 4 [(gogoproto.jsontag) = "search,omitempty"];
+  // Scope is an aditional filter to view requests based on needs review, reviewed, my requests
+  AccessRequestScope Scope = 5 [(gogoproto.jsontag) = "scope,omitempty"];
+  // Requester is the requester of the api call. This is set by the auth server
+  // Use User for the requester of the request.
+  string Requester = 6 [(gogoproto.jsontag) = "requester,omitempty"];
```

Comment:
So this is the identity used in the gRPC client. Right?
I wonder if we really need this value here, or if we could/should extract it from the certificate.

Reply:
I "had" to add it because the matching happens on a filter->request angle and it seemed right to keep this value on the filter. Upcoming change looks like this
```go
func (f *AccessRequestFilter) Match(req AccessRequest) bool {
	// only return if the request was made by the api requester
	if f.Scope == AccessRequestScope_MY_REQUESTS && req.GetUser() != f.Requester {
		return false
	}
	// a user cannot review their own requests
	if f.Scope == AccessRequestScope_NEEDS_REVIEW && req.GetUser() == f.Requester {
		return false
	}
	// only match if the api requester has submit a review
	if f.Scope == AccessRequestScope_REVIEWED {
		reviews := req.GetReviews()
		var reviewers []string
		for _, review := range reviews {
			reviewers = append(reviewers, review.Author)
		}
		if !slices.Contains(reviewers, f.Requester) {
			return false
		}
	}
	if !req.MatchSearch(f.SearchKeywords) {
		return false
	}
	if f.ID != "" && req.GetName() != f.ID {
		return false
	}
	if f.User != "" && req.GetUser() != f.User {
		return false
	}
	if !f.State.IsNone() && req.GetState() != f.State {
		return false
	}
	return true
}
```

where we set this requester only here
```go
func (a *ServerWithRoles) ListAccessRequests(ctx context.Context, req *proto.ListAccessRequestsRequest) (*proto.ListAccessRequestsResponse, error) {
	// ensure filter is non-nil
	if req.Filter == nil {
		req.Filter = &types.AccessRequestFilter{}
	}
	req.Filter.Requester = a.context.User.GetName()
```

Reply:
It can be argued that it doesnt need to be here sure, but if its only being set in the api I don't think its the worst (with the comment stating so as well). I won't die on this hill tho, and will change if you think its required!

Reply:
I'm fine either way :+1: 

Just one last question:
Can this field be used to trick the API into giving someone's else Review Requests, that they wouldn't have access otherwise?

Reply:
even if it was passed from a client, it gets overwritten here by the auth server so I don't think so. I think it'd be equivalent to storing it as a variable and performing the check (albiet not on the request matcher) but this just lets us pass it to the request matcher.

Reply:
Alright, sounds good.
Thank you.